dmriprep/workflows/dwi/util.py

```diff
@@ -292,3 +292,181 @@ def _normalize(in_file, newmax=2000, perc=98.0):
     hdr.set_data_dtype("int16")
     nii.__class__(data.astype("int16"), nii.affine, hdr).to_filename(out_file)
     return out_file
+
+
+def gen_index(in_file):
+    # Generate the index file for eddy
+    import os
+    import numpy as np
+    import nibabel as nib
+    from nipype.utils.filemanip import fname_presuffix
+
+    out_file = fname_presuffix(
+        in_file,
+        suffix="_index.txt",
+        newpath=os.path.abspath("."),
+        use_ext=False,
+    )
+    vols = nib.load(in_file).shape[-1]
+    index_lines = np.ones((vols,))
+    index_lines_reshape = index_lines.reshape(1, index_lines.shape[0])
+    np.savetxt(out_file, index_lines_reshape, fmt="%i")
+    return out_file
+
+
+def gen_acqparams(in_file, total_readout_time=0.05):
+    # Generate the acqp file for eddy
+    import os
+    from ... import config
+    from nipype.utils.filemanip import fname_presuffix
+
+    # Get the metadata for the dwi
+    layout = config.execution.layout
+    metadata = str(layout.get_metadata(in_file))
+
+    # Generate output file name
+    out_file = fname_presuffix(
+        in_file,
+        suffix="_acqparams.txt",
+        newpath=os.path.abspath("."),
+        use_ext=False,
+    )
+
+    # Dictionary for converting directions to acqp lines
+    acq_param_dict = {
+        "j": "0 1 0 %.7f",
+        "j-": "0 -1 0 %.7f",
+        "i": "1 0 0 %.7f",
+        "i-": "-1 0 0 %.7f",
+        "k": "0 0 1 %.7f",
+        "k-": "0 0 -1 %.7f",
+    }
+
+    # Get encoding direction from json
+    if metadata.get("PhaseEncodingDirection"):
+        pe_dir = metadata.get("PhaseEncodingDirection")
+    else:
+        pe_dir = metadata.get("PhaseEncodingAxis")
+
+    # Get readout time from json, use default of 0.05 otherwise
+    if metadata.get("TotalReadoutTime"):
+        total_readout = metadata.get("TotalReadoutTime")
+    else:
+        total_readout = total_readout_time
+
+    # Construct the acqp file lines
```

Comment:
lines? AFAICT, this will generate just one line right?

Reply:
Right, double checked and it generates one line.

Reply:
sdcflows also has a function for getting/calculating this: https://github.com/nipreps/sdcflows/blob/66b3c3009b9d951b19f36902c51e379d36e292f2/sdcflows/utils/epimanip.py#L14

Reply:
Thanks, added sdcflows' function in e3d347cfbad4cccc915953b80e290f4584444f1b